deltacat/compute/converter/pyiceberg/replace_snapshot.py

```diff
@@ -0,0 +1,174 @@
+from typing import Optional, List
+import uuid
+from pyiceberg.table.snapshots import (
+    Operation,
+)
+from pyiceberg.manifest import (
+    DataFileContent,
+    ManifestEntry,
+    ManifestEntryStatus,
+    ManifestFile,
+    write_manifest,
+)
+import itertools
+from pyiceberg.utils.concurrent import ExecutorFactory
+from pyiceberg.table import UpdateSnapshot, _SnapshotProducer
+
+
+class _ReplaceFiles(_SnapshotProducer["_ReplaceFiles"]):
```

Comment:
Can you link the source from which this is copied and is there a reason to copy?